cargo-dist/src/config.rs

```diff
@@ -269,7 +269,10 @@ pub struct DistMetadata {
     /// All of these error out if the required env-vars aren't set. In the future this may
     /// allow for the input to be an array of options to try in sequence.
     #[serde(skip_serializing_if = "Option::is_none")]
-    #[serde(default, with = "opt_string_or_vec")]
+    #[serde(
+        default = "InstallPathStrategy::default_list",
+        with = "opt_string_or_vec"
+    )]
     pub install_path: Option<Vec<InstallPathStrategy>>,
```

Comment:
serde defaults continue to not be the correct way to do defaults in our config system -- this will break config inheritance. Defaults need to be applied where all the other defaults are. Also we probably shouldn't write back this default unless they have a shell/powershell installer. The config is meaningless otherwise.

Reply:
Ahhh, yeah, that makes sense. I hadn't considered config inheritance.